Shopify/shopify-cli
# Fix logout when there are theme permission issues

### WHY are these changes introduced?

Fixes https://github.com/Shopify/shopify-cli/issues/2245

Avoid aborting during `shopify logout` if there are errors while trying to delete the development themes.

### WHAT is this pull request doing?

- Rescue `ShopifyCLI::Abort` and `ShopifyCLI::AbortSilent` when calling `DevelopmentTheme.delete` and do nothing.

### How to test your changes?

Ideally, it should be something like this:
- Log in with a staff account with themes permissions
- `shopify theme init` and `shopify theme serve` to create the development theme
- Remove the theme permissions of the staff user from the owner's dashboard
- `shopify logout`

However, it looks like there's an [issue with the permissions](https://github.com/Shopify/shopify-cli/issues/2245) (maybe cache), so I had to add a `return permission_error` at the beginning of [this method](https://github.com/Shopify/shopify-cli/blob/f994bccdf912c0adf88782345e193061c4d88aef/lib/shopify_cli/theme/theme_admin_api.rb#L35).

### Update checklist
